src/main/scala/org/broadinstitute/dsde/vault/services/analysis/UpdateServiceHandler.scala

```diff
@@ -94,7 +94,7 @@ case class UpdateServiceHandler(requestContext: RequestContext, dmService: Actor
         case true =>
           if (fileCount == bossObjects.size) {
             log.debug("'X-Force-Location' Analysis update complete")
-            requestContext.complete(new Analysis(dmId.get, analysis.input, analysis.metadata, analysis.files))
+            requestContext.complete(new Analysis(dmId.get, analysis.input, analysis.metadata, Option(providedFiles)))
```

Comment:
From running tests on this branch, DM is storing files as a boss id: `bai -> dfdeb1cb-5d15-4d8a-b8fc-506a28e17677` so if you just return what DM hands you, it will not meet the requirement (I think).